rwkv/chat_with_bot.py

```diff
@@ -16,102 +16,43 @@
 
 MAX_GENERATION_LENGTH: int = 250
 
 # Sampling temperature. It could be a good idea to increase temperature when top_p is low.
 TEMPERATURE: float = 0.8
 # For better Q&A accuracy and less diversity, reduce top_p (to 0.5, 0.2, 0.1 etc.)
 TOP_P: float = 0.5
-
-if LANGUAGE == 'English':
-    separator: str = ':'
-
-    if QA_PROMPT:
-        user: str = 'User'
-        bot: str = 'Bot'
-        init_prompt: str = f'''
-The following is a verbose and detailed conversation between an AI assistant called {bot}, and a human user called {user}. {bot} is intelligent, knowledgeable, wise and \
-polite.
-
-{user}{separator} french revolution what year
-
-{bot}{separator} The French Revolution started in 1789, and lasted 10 years until 1799.
-
-{user}{separator} 3+5=?
-
-{bot}{separator} The answer is 8.
-
-{user}{separator} guess i marry who ?
-
-{bot}{separator} Only if you tell me more about yourself - what are your interests?
-
-{user}{separator} solve for a: 9-a=2
-
-{bot}{separator} The answer is a = 7, because 9 - 7 = 2.
-
-{user}{separator} what is lhc
-
-{bot}{separator} LHC is a high-energy particle collider, built by CERN, and completed in 2008. They used it to confirm the existence of the Higgs boson in 2012.
-
-'''
-    else:
-        user: str = 'Bob'
-        bot: str = 'Alice'
-        init_prompt: str = f'''
-The following is a verbose detailed conversation between {user} and a young girl {bot}. {bot} is intelligent, friendly and cute. {bot} is likely to agree with {user}.
-
-{user}{separator} Hello {bot}, how are you doing?
-
-{bot}{separator} Hi {user}! Thanks, I'm fine. What about you?
-
-{user}{separator} I am very good! It's nice to see you. Would you mind me chatting with you for a while?
-
-{bot}{separator} Not at all! I'm listening.
-
-'''
-
-elif LANGUAGE == 'Chinese':
-    separator: str = ':'
-
-    if QA_PROMPT:
-        user: str = 'Q'
-        bot: str = 'A'
-        init_prompt: str = f'''
-Expert Questions & Helpful Answers
-
-Ask Research Experts
-
-'''
-    else:
-        user: str = 'Bob'
-        bot: str = 'Alice'
-        init_prompt: str = f'''
-The following is a verbose and detailed conversation between an AI assistant called {bot}, and a human user called {user}. {bot} is intelligent, knowledgeable, wise and \
-polite.
-
-{user}{separator} what is lhc
-
-{bot}{separator} LHC is a high-energy particle collider, built by CERN, and completed in 2008. They used it to confirm the existence of the Higgs boson in 2012.
-
-{user}{separator} 企鹅会飞吗
-
-{bot}{separator} 企鹅是不会飞的。它们的翅膀主要用于游泳和平衡，而不是飞行。
-
-'''
-else:
-    assert False, f'Invalid language {LANGUAGE}'
+# Penalize new tokens based on whether they appear in the text so far, increasing the model's likelihood to talk about new topics.
+PRESENCE_PENALTY = 0.2
+# Penalize new tokens based on their existing frequency in the text so far, decreasing the model's likelihood to repeat the same line verbatim.
+FREQUENCY_PENALTY = 0.2
```

Comment:
Please add types to all top-level variables, parameters and functions in the file.